libs/full/async_mpi/include/hpx/async_mpi/mpi_future.hpp

```diff
@@ -162,10 +185,10 @@ namespace hpx { namespace mpi { namespace experimental {
 
             // invoke the call to MPI_Ixxx, ignore the returned result for now
             int result = f(std::forward<Ts>(ts)..., &data->request_);
```

Comment:
Reminder: this should handle any return type, including `void`. Edit: I see you have a TODO comment in the code for that. I'm leaving this comment just to keep track of it.